langchain/vectorstores/starrocks.py

```diff
@@ -0,0 +1,447 @@
+
+logger = logging.getLogger()
+DEBUG = False
+
+
+def has_mul_sub_str(s: str, *args: Any) -> bool:
+    for a in args:
+        if a not in s:
+            return False
+    return True
+
+
+def debug_output(s):
+    if DEBUG:
+        print(s)
+    pass
+
+
+def get_named_result(connection, query):
+    import pymysql.cursors
+    cursor: pymysql.cursors.Cursor = connection.cursor()
+    cursor.execute(query)
+    columns = cursor.description
+    result = []
+    for value in cursor.fetchall():
+        r = {}
+        for idx, datum in enumerate(value):
+            k = columns[idx][0]
+            r[k] = datum
+        result.append(r)
+    debug_output(result)
+    cursor.close()
+    return result
+
+
+class StarRocksSettings(BaseSettings):
+    """StarRocks Client Configuration
+
+    Attribute:
+        StarRocks_host (str) : An URL to connect to MyScale backend.
+                             Defaults to 'localhost'.
+        StarRocks_port (int) : URL port to connect with HTTP. Defaults to 8443.
+        username (str) : Username to login. Defaults to None.
+        password (str) : Password to login. Defaults to None.
+        database (str) : Database name to find the table. Defaults to 'default'.
+        table (str) : Table name to operate on.
+                      Defaults to 'vector_table'.
+
+        column_map (Dict) : Column type map to project column name onto langchain
+                            semantics. Must have keys: `text`, `id`, `vector`,
+                            must be same size to number of columns. For example:
+                            .. code-block:: python
+
+                                {
+                                    'id': 'text_id',
+                                    'embedding': 'text_embedding',
+                                    'document': 'text_plain',
+                                    'metadata': 'metadata_dictionary_in_json',
+                                }
+
+                            Defaults to identity map.
+    """
+
+    host: str = "localhost"
+    port: int = 9030
+
+    username: Optional[str] = None
+    password: Optional[str] = None
+
+    column_map: Dict[str, str] = {
+        "id": "id",
+        "document": "document",
+        "embedding": "embedding",
+        "metadata": "metadata",
+    }
+
+    database: str = "default"
+    table: str = "langchain"
+
+    def __getitem__(self, item: str) -> Any:
+        return getattr(self, item)
+
+    class Config:
+        env_file = ".env"
+        env_prefix = "starrocks_"
+        env_file_encoding = "utf-8"
+
+
+class StarRocks(VectorStore):
+    """Wrapper around StarRocks vector database
+
+    You need a `pymysql` python package, and a valid account
+    to connect to StarRocks.
+    """
+
+    def __init__(
+            self,
+            embedding: Embeddings,
+            config: Optional[StarRocksSettings] = None,
+            **kwargs: Any,
+    ) -> None:
+        """StarRocks Wrapper to LangChain
```

Comment:
better docstring would be nice

Reply:
I've updated  docstring and added a example notebook to use starrocks as vectordb.